dhall/src/Dhall/Pretty/Internal.hs-boot

```diff
@@ -1,13 +1,27 @@
 module Dhall.Pretty.Internal where
 
+import Control.DeepSeq (NFData)
+import Data.Data (Data)
 import Data.Text (Text)
 import Data.Text.Prettyprint.Doc (Pretty, Doc)
 import Dhall.Src (Src)
+import Language.Haskell.TH.Syntax (Lift)
 
 import {-# SOURCE #-} Dhall.Syntax
 
 data Ann
 
+data CharacterSet = ASCII | Unicode
+
+instance Eq CharacterSet
+instance Ord CharacterSet
+instance Show CharacterSet
+instance Data CharacterSet
+instance Lift CharacterSet
+instance NFData CharacterSet
+instance Semigroup CharacterSet
+instance Monoid CharacterSet
```

Comment:
I am not 100% sure how `hs-boot` files work. I had write the constructors and the instances otherwise they wouldn't propagate properly. Is there another way?

Reply:
What you did is fine, but to answer your question: another solution is to factor out the `CharacterSet` type into its own module (e.g. `Dhall.CharacterSet`) to break the cyclic dependency